orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -267,9 +267,14 @@ def _initialize_graph_builder(self, training):
         """Creates a new OrtModuleGraphBuilder, initializes it and saves it to self._graph_builder"""
 
         # TODO: PyTorch exporter bug: changes the initializer order in ONNX model
-        initializer_names = [name for name, _ in self._flattened_module.named_parameters()]
-        initializer_names_to_train = [name for name,
-                                      param in self._flattened_module.named_parameters() if param.requires_grad]
+        # All initializer names along with user inputs are a part of the onnx graph inputs
+        # since the onnx model was exported with the flag keep_initializers_as_inputs=True
+        onnx_initializer_names = {p.name for p in self._onnx_model.graph.input}
+
+        initializer_names = [name for name, _ in self._flattened_module.named_parameters()
+                             if name in onnx_initializer_names]
+        initializer_names_to_train = [name for name, param in self._flattened_module.named_parameters()
+                                      if param.requires_grad and name in onnx_initializer_names]
```

Comment:
I am not sure if printing a warning here is a good idea. Consider an example:
```py
def forward(self, bool_input, input1):
    if bool_input:
        do_something_with_fc1
    else:
        do_somethind_else_with_fc2
```

For this example, whether the user enters ```bool_input``` as ```True``` or ```False```, outputting a warning might be informative, but also unnecessary. In what scenario do you think it would be detrimental to the user with a warning not printed?